src/core/toga/widgets/table.py

```diff
@@ -25,127 +20,168 @@ def __repr__(self):
         return node
 
     def remove(self, node):
         self._data.remove(node)
-        if self.interface:
-            self.interface._impl.remove_node(node)
+        self._refresh()
+
+    def item(self, row: int, column: int):
+        if isinstance(row and column, int):
+            return self._data[row].data[column]
+
+    def row(self, row: int) -> TableRow:
+        if row >= 0:
+            return self._data[row]
 
+    @property
+    def rows(self) -> list:
+        return self.data
 
 
 class Table(Widget):
-    """ A Table Widget allows the disply of data in the from of columns and rows.
+    """ A Table Widget allows the display of data in the from of columns and rows.
 
     Args:
         headings (``list`` of ``str``): The list of headings for the table.
         id (str): An identifier for this widget.
+        data (``list`` of ``tuple``): The data to be displayed on the table.
         style (:class:`colosseum.CSSNode`): An optional style object.
-            If no style is provided then a new one will be created for the widget.
+            If no style is provided` then a new one will be created for the widget.
+        on_select (``callable``): A function to be invoked on selecting a row of the table.
         factory (:obj:`module`): A python module that is capable to return a
             implementation of this class with the same name. (optional & normally not needed)
+
+    Examples:
+        >>> headings = ['Head 1', 'Head 2', 'Head 3']
+        >>> data = [('item 1', 'item 2', 'item3'),
+        >>>         ('item 1', 'item 2', 'item3')]
+        >>>
+        >>> table = Table(headings, data=data)
     """
+
     def __init__(self, headings, id=None, style=None, data=None, on_select=None, factory=None):
         super().__init__(id=id, style=style, factory=factory)
         self.headings = headings
-        self._data = data
+        self._data = None
         self._impl = self.factory.Table(interface=self)
+        self.data = data
 
         self.on_select = on_select
 
     @property
     def data(self):
-        '''
-        :returns: The data source of the tree
-        :rtype: ``dict``
-        '''
-        return self._data
+        """ The data source of the widget. It accepts table data
+        in the form of ``list``, ``tuple``, or :obj:`ListDataSource`
+
+        Returns:
+            Returns a (:obj:`ListDataSource`).
+        """
+        return self._data if self._data is not None else None
 
     @data.setter
     def data(self, data):
-        '''
-        Set the data source of the data
-
-        :param data: Data source
-        :type  data: ``list``, ``tuple``, or ``class``
-        '''
         if isinstance(data, (list, tuple)):
             self._data = ListDataSource(data)
-        else:
+        elif isinstance(data, ListDataSource):
             self._data = data
+        else:
+            raise UserWarning('The table widgets should get data in form of a list or ListDataSource.')
```

Comment:
We definitely don't want to be this prescriptive. The old construction would have allowed any user-created class that quacked the same as a ListDataSource to be used as a data source.